haskell/private/actions/compile.bzl

```diff
@@ -175,7 +174,36 @@ def _process_chs_file(hs, cc, ghc_defs_dump, chs_file, chi_files=[]):
 
   return hs_out, chi_out, idir
 
-def _compilation_defaults(hs, cc, java, dep_info, srcs, extra_srcs, cpp_defines, compiler_flags, main_file = None, my_pkg_id = None):
+def _output_file_ext(base, dynamic, profiling_enabled):
```

Comment:
I find this function doesn't pay its own way and could be more harmful than good. The cheap option is to construct the file name extensions manually every time: `.p_o_dyn` etc. It should prove more readable. Also your line breaking below won't need to be as ugly, because the `declare_compiled` stuff will fit on one line.

Reply:
Yes and no... I think it reduces the possibility of typos somewhat, but the main reason is the `with_profiling` argument. If we "inline" `_output_file_ext` (with some beta-reduction) there will be a lot of conditionals of the form:

```
"." + ("_p" if with_profiling else "") + "dyn_o"
```

Etc. which is not much better and definitely not shorter. I'd leave the function "as is".

(Well another option would be something like this: `".p_dyn_o" if with_profiling else ".dyn_o"` which is typo-prone I think and not really shorter either.)